client/sidebar/header/CreateChannel.js

```diff
@@ -0,0 +1,265 @@
+import { FlowRouter } from 'meteor/kadira:flow-router';
+import React, { memo, useCallback, useMemo, useState } from 'react';
+import { useMutableCallback, useDebouncedCallback } from '@rocket.chat/fuselage-hooks';
+import { Box, Modal, ButtonGroup, Button, TextInput, Icon, Field, ToggleSwitch } from '@rocket.chat/fuselage';
+
+import { useTranslation } from '../../contexts/TranslationContext';
+import { useForm } from '../../hooks/useForm';
+import { useEndpointActionExperimental } from '../../hooks/useEndpointAction';
+import UserAutoCompleteMultiple from '../../../ee/client/audit/UserAutoCompleteMultiple';
+import { useSetting } from '../../contexts/SettingsContext';
+import { usePermission } from '../../contexts/AuthorizationContext';
+import { useMethod } from '../../contexts/ServerContext';
+import { useComponentDidUpdate } from '../../hooks/useComponentDidUpdate';
+
+
+export const CreateChannel = ({
```

Comment:
Thanks for making the component generic!!! It'll help a lot on a future task.

Reply:
thanks man, congratulations from the team: me and @MartinSchoeler